src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,560 @@
+Object.assign(pc, function () {
+
+    var ANIM_INTERRUPTION_SOURCE_NONE = 'NONE';
+    var ANIM_INTERRUPTION_SOURCE_PREV_STATE = 'PREV_STATE';
+    var ANIM_INTERRUPTION_SOURCE_NEXT_STATE = 'NEXT_STATE';
+    var ANIM_INTERRUPTION_SOURCE_PREV_STATE_NEXT_STATE = 'PREV_STATE_NEXT_STATE';
+    var ANIM_INTERRUPTION_SOURCE_NEXT_STATE_PREV_STATE = 'NEXT_STATE_PREV_STATE';
+
+    var ANIM_TRANSITION_PREDICATE_GREATER_THAN = 'GREATER_THAN';
+    var ANIM_TRANSITION_PREDICATE_LESS_THAN = 'LESS_THAN';
+    var ANIM_TRANSITION_PREDICATE_GREATER_THAN_EQUAL_TO = 'GREATER_THAN_EQUAL_TO';
+    var ANIM_TRANSITION_PREDICATE_LESS_THAN_EQUAL_TO = 'LESS_THAN_EQUAL_TO';
+    var ANIM_TRANSITION_PREDICATE_EQUAL_TO = 'EQUAL_TO';
+    var ANIM_TRANSITION_PREDICATE_NOT_EQUAL_TO = 'NOT_EQUAL_TO';
+
+    var ANIM_PARAMETER_INTEGER = 0;
+    var ANIM_PARAMETER_FLOAT = 1;
+    var ANIM_PARAMETER_BOOLEAN = 2;
+    var ANIM_PARAMETER_TRIGGER = 3;
+
+    var ANIM_PARAMETER_TYPE_NAMES = {
+        ANIM_PARAMETER_INTEGER: 'Integer',
+        ANIM_PARAMETER_FLOAT: 'Float',
+        ANIM_PARAMETER_BOOLEAN: 'Boolean',
+        ANIM_PARAMETER_TRIGGER: 'Trigger'
+    };
+
+    var ANIM_STATE_START = 'ANIM_STATE_START';
+    var ANIM_STATE_END = 'ANIM_STATE_END';
+
+    var AnimState = function (name, speed) {
+        this.name = name;
+        this.animations = [];
+        this.speed = speed || 1.0;
+    };
+
+    Object.assign(AnimState.prototype, {
+        isPlayable: function () {
```

Comment:
Can we make all of these functions getter properties instead? That way this API would be more consistent e.g. with more recent components like the sound.

Reply:
Yep, agreed.

Reply:
I've moved all of these over to getters now.